Scripts/OnPrem/certificateChange.sh

```diff
@@ -0,0 +1,273 @@
+        echo "    Executing: sudo mv $INSIGHTS_DIR/cert.pfx $INSIGHTS_DIR/cert.pfx_$randomN"
+        sudo mv $INSIGHTS_DIR/cert.pfx $INSIGHTS_DIR/cert.pfx_$randomN
+        if [ $? -ne 0 ]; then
+            echo "FATAL: Could not move file"
+            echo "Cleaning up /tmp/tmp.pem"
+            rm "/tmp/tmp.pem"
+            exit 1
+        fi
+    fi
+
+    echo "  Creating new cert.pfx"
+    echo "    Executing: sudo docker exec -it looker-container openssl pkcs12 -export -in /app/.deploy/tmp.pem -out /app/.deploy/cert.pfx -passout pass:"
+    sudo docker exec -it looker-container openssl pkcs12 -export -in /app/.deploy/tmp.pem -out /app/.deploy/cert.pfx -passout pass:
+
+    if [[ $? -ne 0 ]]; then
+        echo "Fatal: Failed to create $INSIGHTS_DIR/cert.pfx"
+        echo "Cleaning up $INSIGHTS_DIR/tmp.pem"
+        rm "$INSIGHTS_DIR/tmp.pem"
+        exit 1
+    fi 
+    
+    echo "    Executing: sudo chmod 744 $INSIGHTS_DIR/cert.pfx"
+    sudo chmod 744 $INSIGHTS_DIR/cert.pfx
+    
+    if [[ $? -ne 0 ]]; then 
+       echo "Fatal: Failed to set perms on $INSIGHTS_DIR/cert.pfx" 
+       exit 1
+    fi
+    
+    
+    echo "  Successfully created $INSIGHTS_DIR/cert.pfx"
+    echo ""
+
+
+}
+
+
+# Check if a specific docker container is running
+function check_docker_container() {
+
+    echo "Checking that the docker container is running: sudo  docker ps -q -f name=looker-container"
+    if [[ -z $(sudo docker ps -q -f name=looker-container) ]]; then
+        echo "FATAL: Docker container $CONTAINER_NAME is not running."
+        echo "FATAL: Try: sudo systemctl restart docker"
+        exit 1
+    fi
+
+    echo "  Docker container is running"
+    echo ""
+}
+
+# Fine the location of insights
+function set_insights_dir() {
+    echo "Setting insights working directory"
+    echo "  Executing: sudo docker inspect looker-container | jq -r '.[0].HostConfig.Binds' | grep 'deploy' | awk -F':/app/.deploy' '{print $1}' | awk -F'\"' '{print $2}'"
+    INSIGHTS_DIR=$(sudo docker inspect looker-container | jq -r '.[0].HostConfig.Binds' | grep 'deploy' | awk -F':/app/.deploy' '{print $1}' | awk -F'"' '{print $2}')
+    
+    if [ $? -ne 0 ]; then
+        echo "FATAL: Could not set the insights working directory"
+        echo "FATAL: Try running command manually to see the error"
+        exit 1
+    fi
+
+    echo "  Verifying that $INSIGHTS_DIR exists"
+    if [[ ! -e $INSIGHTS_DIR ]]; then
+        echo "FATAL: Insights directory is missing!"
+        echo "FATAL: Open a ticket with UiPath to repair this"
+        exit 1
+    fi 
+
+    echo ""
+}
+
+# Run the certificate update
+function run_certificate_update() {
+    echo "Running certificate update"
+    echo "  Executing: sudo docker exec -it looker-container su looker -m -c '</app/looker-init-job/install-certificate.sh bash'"
+
+   sudo docker exec -it looker-container su looker -m -c '</app/looker-init-job/install-certificate.sh bash'
```

Comment:
if the cert.pfx is the same, the `install-certificate.sh` will not update
![image](https://github.com/UiPath/Insights-Customer/assets/76539124/4b790957-79d2-42bf-822b-940bf4e64585)

to force "reinstall" you can delete the "/app/workdir/.ssl/cert.pfx"

Reply:
When we create the certificate, we overwrite the one that is there. Look at convert_pfx. If cert.pfx exists it moves and and then over writes it.